### PR TITLE
Fix #7

### LINKED ISSUE #7: allow using keystore as a backend
**Description**
allow using keystore as a backend, currently we have only plain text json backend available.
 
**Use case or motivation behind the feature request**
encrypted keystore makes it harder to extract credentials from the system.
 
**Related issues**

 
**Additional context**
customer request.

see also [stackoverflow](https://stackoverflow.com/questions/6243446/how-to-store-a-simple-key-string-inside-java-keystore)

### PR DESCRIPTION
**Pull request by kortemik**

### CODE REVIEW

File: src/main/java/com/teragrep/jai_02/KeyStoreCredentialLookup.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Enumeration;
+
+public class KeyStoreCredentialLookup {
```
Comment (kortemik) [inline src/main/java/com/teragrep/jai_02/KeyStoreCredentialLookup.java:60]: this must be worked into object(s) with clear responsibilities.

File: src/test/java/com/teragrep/jai_02/tests/KeyStoreCredentialLookupTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableEntryException;
+import java.security.cert.CertificateException;
+import java.security.spec.InvalidKeySpecException;
+
+public class KeyStoreCredentialLookupTest {
```
Comment (kortemik) [inline src/test/java/com/teragrep/jai_02/tests/KeyStoreCredentialLookupTest.java:62]: this must be worked into proper object tests.

**Comment by kortemik** [comment]
there should be non-equality tests as well.

**Comment by eemhu** [comment]
looks good

**Review by eemhu — APPROVED** [review]

**Pull request closed without merging**